caddyhttp/proxy/reverseproxy.go

```diff
@@ -226,15 +230,21 @@ func NewSingleHostReverseProxy(target *url.URL, without string, keepalive int) *
 		}
 	}
 
+	var dialer = *defaultDialer
```

Comment:
This is strange syntax to me, let's use `dialer := *defaultDialer` instead (also, might be worth noting that this isn't a deep copy, but I guess it works in this case)